packages/headless/src/features/templates/templates-manager.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  ArrayValue,
+  NumberValue,
+  Schema,
+  SchemaValidationError,
+  Value,
+} from '@coveo/bueno';
+import {requiredNonEmptyString} from '../../utils/validate-payload';
+
+export type TemplateCondition<ItemType> = (item: ItemType) => boolean;
+
+export interface Template<ItemType, Content = unknown> {
+  /**
+   * The stored content of the template.
+   */
+  content: Content;
+  /**
+   * A list of conditions that must be fulfilled for this template to be selected.
+   */
+  conditions: TemplateCondition<ItemType>[];
+  /**
+   * A value which the manager will fallback to when multiple templates' conditions are fulfilled.
+   * Templates with higher priority values will be selected over others. The minimum value is `0`.
+   */
+  priority?: number;
+  /**
+   * A list of index fields that are necessary to render the template.
+   */
+  fields?: string[];
+}
+
+export const templateSchema = new Schema({
+  content: new Value({required: true}),
+  conditions: new Value({required: true}),
+  priority: new NumberValue({required: false, default: 0, min: 0}),
+  fields: new ArrayValue({
+    required: false,
+    each: requiredNonEmptyString,
+  }),
+});
+
+export interface TemplatesManager<ItemType, TemplateContent = unknown> {
```

Comment:
```suggestion
export interface TemplateManager<ItemType, TemplateContent = unknown> {
```

Reply:
It is managing multiple templates though...
And the old interface was(and still is) `ResultTemplatesManager` so I would not change this one, so that it fit with what gets exported 🤷